### PR TITLE
Adding the ability to skip hdfs validation

### PR DESCRIPTION
**Pull request by znichols**

### CODE REVIEW

**Comment by nevillelyh** [comment]
We want to avoid creating too many variations of these classes and should add a withoutValidation method like other IO classes. Can you keep it in your repo for now?

File: scio-hdfs/src/main/java/com/google/cloud/dataflow/sdk/io/hdfs/skipvalidation/SkipValidationHDFSFileSink.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * A {@code Sink} for writing records to a Hadoop filesystem using a Hadoop file-based output
+ * format with Simple Authentication.
+ *
+ * Skips validation for hdfs location (in case the permissions don't exist to read it on the
+ * calling machine)
+ *
+ * @param <K> The type of keys to be written to the sink.
+ * @param <V> The type of values to be written to the sink.
+ */
+public class SkipValidationHDFSFileSink<K, V> extends HDFSFileSink<K, V> {
```
Comment (ravwojdyla) [inline scio-hdfs/src/main/java/com/google/cloud/dataflow/sdk/io/hdfs/skipvalidation/SkipValidationHDFSFileSink.java]: Why is this not part of `HDFSFileSink` itself?

File: scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala [hunk 2]
```diff
@@ -207,7 +208,8 @@ package object hdfs {
     // TODO: numShards
     def saveAsHdfsTextFile(path: String,
                            username: String = null,
-                           conf: Configuration = null): Future[Tap[String]] = {
+                           conf: Configuration = null,
+                           skipValidation: Boolean = false): Future[Tap[String]] = {
```
Comment (ravwojdyla) [inline scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala]: `validate: Boolean = true` ?

File: scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala [hunk 3]
```diff
@@ -207,7 +208,8 @@ package object hdfs {
     // TODO: numShards
     def saveAsHdfsTextFile(path: String,
```
Comment (ravwojdyla) [inline scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala]: What about `Avro`?

**Comment by ravwojdyla** [comment]
@nevillelyh technically Dataflow SDK has validation option - scio currently does not expose it for sinks, maybe we should in fact.

**Comment by ravwojdyla** [comment]
@nevillelyh actually after some thought - I agree - 👎

**Comment by nevillelyh** [comment]
@znichols closing this for now but I'll work on refactoring HDFS code soon.

**Comment by znichols** [comment]
SGTM, thanks guys

**Pull request closed without merging**